File: ios/Plugin/OpenWithPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(OpenWithPlugin, "OpenWith",
    CAP_PLUGIN_METHOD(addHandler, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(setVerbosity, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(getVerbosity, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(load, CAPPluginReturnPromise);
) 

File: ios/Sources/OpenWithPlugin/OpenWithPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(OpenWithPlugin)
public class OpenWithPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "OpenWithPlugin"
    public let jsName = "OpenWith"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = OpenWith()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
